server/data/licenceApiClient.ts

```diff
@@ -15,79 +14,109 @@ import type {
 } from '../@types/licenceApiClientTypes'
 import config, { ApiConfig } from '../config'
 import LicenceStatus from '../enumeration/licenceStatus'
+import { User } from '../@types/CvlUserDetails'
 
-export default class LicenceApiClient {
-  restClient: RestClient
-
-  constructor(token: string) {
-    this.restClient = new RestClient('Licence API', config.apis.licenceApi as ApiConfig, token)
-  }
-
-  async getTestData(): Promise<LicenceApiTestData[]> {
-    return (await this.restClient.get({ path: `/test/data` })) as Promise<LicenceApiTestData[]>
+export default class LicenceApiClient extends RestClient {
+  constructor() {
+    super('Licence API', config.apis.licenceApi as ApiConfig)
   }
 
-  async createLicence(licence: CreateLicenceRequest): Promise<LicenceSummary> {
-    return (await this.restClient.post({
-      path: `/licence/create`,
-      data: licence,
-    })) as Promise<LicenceSummary>
+  async createLicence(licence: CreateLicenceRequest, user: User): Promise<LicenceSummary> {
+    return (await this.post(
+      {
+        path: `/licence/create`,
+        data: licence,
+      },
+      { username: user.username }
```

Comment:
Example here of an API client choosing to supply the username from the User object rather than the token

Reply:
Like it. Though, as described above, not sure about overloading the CvlUserDetails class with Express User - could be a separate, much more simple object? Argue it though, as I'm not 100% - there might be other reasons for doing so!

Reply:
Lets leave it like it is for now. I can see that the User object is a convenient vehicle to pass around, containing everything that services and clients need.